forge/src/main/java/gg/chaldea/client/reset/packet/ClientReset.java

```diff
@@ -1,116 +1,185 @@
-
-        context.setPacketHandled(true);
-        FMLNetworkConstants.handshakeChannel.reply(
-                new FMLHandshakeMessages.C2SAcknowledge(),
-                new NetworkEvent.Context(connection, NetworkDirection.LOGIN_TO_CLIENT, 98)
-        );
-
-        FMLHandshakeHandler.LOGGER.debug(FMLHandshakeHandler.FMLHSMARKER, "Reset complete");
-    }
-
-    @OnlyIn(Dist.CLIENT)
-    public static boolean handleClear(NetworkEvent.Context context) {
-        CompletableFuture<Void> future = context.enqueueWork(() -> {
-            FMLHandshakeHandler.LOGGER.debug(FMLHandshakeHandler.FMLHSMARKER, "Clearing");
-
-            // Preserve
-            ServerData serverData = Minecraft.getInstance().getCurrentServer();
-
-            // Clear
-            if (Minecraft.getInstance().level == null) {
-                // Ensure the GameData is reverted in case the client is reset during the handshake.
-                GameData.revertToFrozen();
-            }
-
-            // Clear
-            Minecraft.getInstance().clearLevel(new DirtMessageScreen(new TranslationTextComponent("connect.negotiating")));
-
-            // Restore
-            Minecraft.getInstance().setCurrentServer(serverData);
-        });
-
-        FMLHandshakeHandler.LOGGER.debug(FMLHandshakeHandler.FMLHSMARKER, "Waiting for clear to complete");
-        try {
-            future.get();
-            FMLHandshakeHandler.LOGGER.debug("Clear complete, continuing reset");
-            return true;
-        } catch (Exception ex) {
-            FMLHandshakeHandler.LOGGER.error(FMLHandshakeHandler.FMLHSMARKER, "Failed to clear, closing connection", ex);
-            context.getNetworkManager().disconnect(new StringTextComponent("Failed to clear, closing connection"));
-            return false;
-        }
-    }
+	public static final Field handshakeField;
+	public static final Constructor contextConstructor;
+	static final Logger logger = LogManager.getLogger();
+	static final Marker RESETMARKER = MarkerManager.getMarker("RESETPACKET").setParents(MarkerManager.getMarker("FMLNETWORK"));
+
+	public static SimpleChannel handshakeChannel;
+
+	public ClientReset() {
+		IEventBus bus = FMLJavaModLoadingContext.get().getModEventBus();
+		bus.addListener(ClientReset::init);
+		ModLoadingContext.get().registerExtensionPoint(DisplayTest.class, () -> new DisplayTest(() -> NetworkConstants.IGNORESERVERONLY, (a, b) -> true));
+	}
+
+	@SubscribeEvent
+	public static void init(FMLCommonSetupEvent event) {
+		if (handshakeField == null) {
+			logger.error(RESETMARKER, "Failed to find FML's handshake channel. Disabling mod.");
+			return;
+		}
+		if (contextConstructor == null) {
+			logger.error(RESETMARKER, "Failed to find FML's network event context constructor. Disabling mod.");
+			return;
+		}
+		try {
+			//handshakeField.setAccessible(true);
+			//contextConstructor.setAccessible(true);
+			Object handshake = handshakeField.get(null);
+			if (handshake instanceof SimpleChannel) {
```

Comment:
Why would we not throw an error if it's not an instance of simple channel, because it won't throw an exception so the catch clause would not be reached